config/static.c

```diff
@@ -26,103 +35,121 @@
 #include "gatekeeper_lls.h"
 #include "gatekeeper_rt.h"
 
+#define RTE_LOGTYPE_CONFIG RTE_LOGTYPE_USER2
+
+/* TODO Get the install-path via Makefile. */
+#define LUA_BASE_DIR "./lua/"
+#define LUA_CONFIG_FILE     "gatekeeper_config.lua"
+
+struct gatekeeper_config {
+	struct bp_config 	*bp_conf;
+	struct catcher_config 	*catcher_conf;
+	struct dynamic_config 	*dy_conf;
+	struct cps_config 	*cps_conf;
+	struct ggu_config 	*ggu_conf;
+	struct gk_config 	*gk_conf;
+	struct gt_config 	*gt_conf;
+	struct rt_config 	*rt_conf;
+};
+
+static lua_State *lua_state;
```

Comment:
The case you're considering will be the exception, and only a couple of resources fall under this case. For these cases, one can simply use reference counting (https://en.wikipedia.org/wiki/Reference_counting).

Notice that reference counting was already necessary in the Link Layer Support block in order to  know when nobody is interested in a given map.